src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java

```diff
@@ -39,7 +45,17 @@ public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException
         try {
             DSTableInterface<Vehicle> dbResponse = vehicleRepository.findById(id)
                     .orElseThrow(() -> new VehicleNotFoundException("No vehicle with ID : " + id + " was found"));
-            return VehicleValueMapper.entityToResponse(dbResponse.createEntity());
+            VehicleResponse vehicleResponse = VehicleValueMapper.entityToResponse(dbResponse.createEntity());
+            // Check request parameters
+            if (info.equals("location")) {
```

Comment:
This solution is not scalable. More than one type of info can be requested. Please make changes to accomodate this and change the if to a switch case

Reply:
It's now replaced by switch case